api/services/setup.go

```diff
@@ -88,8 +79,9 @@ func (s *service) Setup(ctx context.Context, req requests.Setup) error {
 		},
 	}
 
+	// TODO: use a transaction here
```

Comment:
I understand that you want to do the user creation and the namespace creation using a database transaction and avoid the need to create and delete the user just by logic. However, I think that this to-do is likely to be left for the future because it's going to demand some more complicated changes in the datastore layer. So my suggestion is for you to describe it better in the description so we can leave it for the next step of the work.